codegen/templates/vec.rs.tera

```diff
@@ -421,10 +421,10 @@ impl {{ self_t }} {
 {% elif dim == 4 %}
     /// Creates a 2D vector from the `x`, `y` and `z` elements of `self`, discarding `w`.
     ///
-    /// Truncation to `{{ vec3_t }}` may also be performed by using `self.xyz()` or `{{ vec3_t }}::from()`.
+    /// Truncation to [`{{ vec3_t }}`] may also be performed by using [`self.xyz()`][crate::swizzles::Vec4Swizzles::xyz()].
 {%- if scalar_t == "f32" %}
     ///
-    /// To truncate to `Vec3A` use `Vec3A::from()`.
+    /// To truncate to [`Vec3A`] use [`Vec3A::from()`].
```

Comment:
This isn't linking to the right type, we should probably have a `<Vec3A as From<{{ vec4_t }}>>::from()` for that but it is super verbose and doesn't work.

Reply:
Perhaps a limitation in rustdoc?

Reply:
Wouldn't be surprised, `<X as Y>::...` is rather complex to parse, and I don't think there's another easy way to point out exactly which trait `impl` _and trait `impl` `fn`_ is intended. Clicking the links on the latest docs.rs show that the `impl` is a relatable unique value, but the `from` just has a numeric suffix:

`impl From<Vec3> for Vec3A`:
https://docs.rs/glam/latest/glam/f32/struct.Vec3A.html#impl-From%3CVec3%3E-for-Vec3A
https://docs.rs/glam/latest/glam/f32/struct.Vec3A.html#method.from-8


`impl From<Vec3A> for Vec3`:
https://docs.rs/glam/latest/glam/f32/struct.Vec3A.html#impl-From%3CVec3A%3E-for-Vec3
https://docs.rs/glam/latest/glam/f32/struct.Vec3A.html#method.from

Don't think it's worth spending much time figuring this out, fully-qualified syntax seems to be the only way to solve this and that's not yet supported:
https://github.com/rust-lang/rust/issues/74563
https://rust-lang.github.io/rfcs/1946-intra-rustdoc-links.html#linking-to-associated-items
https://rust-lang.github.io/rfcs/1946-intra-rustdoc-links.html#traits-in-scope